File: framework/test2/common/intersect/Intersect.java
import soda.unittest.*;

import java.util.*;
import java.util.function.Supplier;
import java.util.stream.*;

import soda.leetcode.*;
import soda.unittest.*;


class Solution {
    public List<Integer> intersection(int[] nums1, int[] nums2) {
        if (nums1.length > nums2.length) {
            return intersection(nums2, nums1);
        }
        var mset = new HashSet<Integer>();
        var res = new HashSet<Integer>();
        for (int n : nums1) {
            mset.add(n);
        }
        for (int b : nums2) {
            if (mset.contains(b)) {
                res.add(b);
            }
        }
        return new ArrayList<>(res);
    }
}

public class Intersect implements Supplier<TestWork> {

    @Override
    public TestWork get() {
        var work = new TestWork(new Solution(), "intersection");
        // var work = TestWork.forStruct(Struct.class);
        work.setValidator(Validators.forList(Integer.class, false));
        // work.setCompareSerial(true);
        return work;
    }

    public static void main(String[] args) throws Exception {
        new Intersect().get().run();
    }

}

File: framework/test2/common/intersect/Intersect.cs
using Soda.Unittest;

public class Solution {
    public List<int> Intersection(int[] nums1, int[] nums2) {
        if (nums1.Length > nums2.Length) {
            return Intersection(nums2, nums1);
        }
        var mset = new HashSet<int>();
        var res = new HashSet<int>();
        foreach (int n in nums1) {
            mset.Add(n);
        }
        foreach (int b in nums2) {
            if (mset.Contains(b)) {
                res.Add(b);
            }
        }
        return res.ToList();
    }
}

public class Intersect
{
    public static void Main(string[] args)
    {
        var work = WorkFactory.Create(Utils.Fn(new Solution().Intersection));
        work.SetValidator(Validators.ForList<int>(false));
        // work.CompareSerial = true;
        Console.WriteLine(work.Run(Utils.readStdin()));
    }
}
